stdpopsim/arabidopsis_thaliana.py

```diff
@@ -0,0 +1,113 @@
+"""
+Genome, genetic map and demographic model definitions for Arabidopsis thaliana.
+"""
+import msprime
+import numpy as np
+
+import stdpopsim.models as models
+import stdpopsim.genomes as genomes
+import stdpopsim.genetic_maps as genetic_maps
+
+
+###########################################################
+#
+# Genetic maps
+#
+###########################################################
+
+
+class Salome2012(genetic_maps.GeneticMap):
+    """
+    Genetic map from Salome 2012 averaged across population crosses.
+    Please see this repo for details on how this was done: https://github.com/
+    LohmuellerLab/arabidopsis_recomb_maps
+
+    """
+
+    url = ("http://www.eeb.ucla.edu/Faculty/Lohmueller/data/"
+           "uploads/salome2012_maps.tar.gz")
+    file_pattern = "arab_{name}_map_loess.txt"
+
+
+genetic_maps.register_genetic_map(Salome2012())
+
+
+###########################################################
+#
+# Genome definition
+#
+###########################################################
+
+# Data for length information based on:
+# https://www.arabidopsis.org/portals/genAnnotation/
+#   gene_structural_annotation/agicomplete.jsp
+# Lengths from TAIR 10 although its unclear what reference the genetic map used
+#   -- follow up on this with Salome 2012 authors
+
+_chromosome_data = """\
+chr1 30427671
+chr2 19698289
+chr3 23459830
+chr4 18585056
+chr5 26975502
+"""
+# mutation rate from Ossowski 2010 Science
+# recombination value from Huber et al 2014 MBE
+# rho=200/Mb, assume Ne=124,000, rho=2*Ne*r
+_chromosomes = []
+for line in _chromosome_data.splitlines():
+    name, length = line.split()[:2]
+    _chromosomes.append(genomes.Chromosome(
+        name=name, length=int(length),
+        mean_mutation_rate=7e-9,
+        mean_recombination_rate=8.1e-9))
+
+genome = genomes.Genome(
+    species="arabidopsis_thaliana",
+    chromosomes=_chromosomes,
+    default_genetic_map=Salome2012.name)
+
+
+###########################################################
+#
+# Demographic models
+#
+###########################################################
+
+
+class Durvasula2017MSMC(models.Model):
+    """
+    Model estimated from two homozygous individuals from the South Middle Atlas
+    using MSMC
```

Comment:
Maybe some more information here: how are the `.times` and `.sizes` related? E.g., is the effective population size in time interval `.times[k]` to `times[k+1]` equal to `.sizes[k]`?  With the last size covering `.times[n]` to infinity?

Reply:
It looks from the code like the pop size is `.sizes[0]` on `[0, .times[0])` but also on `[times[0], times[1])`; is this a bug?

Reply:
Thanks Andy! Just added a comment about that. This behavior is because MSMC's output can produce inaccurate estimates in the very recent and very ancient past (e.g. http://www.g3journal.org/content/7/11/3605). Instead of having users simulate 40 million arabidopsis individuals in the most recent generations, we thought it best to use the population size from the first time bin we are confident in and extend that to the present (and likewise for the ancient time bins). So the first 8 time bins have the same population size and the last 3 do as well.

I'm happy to discuss this point further though -- it could be that what we did isn't what we want to have the default model doing. We discuss this point in the supplement of the 2017 paper (line 336) https://www.pnas.org/content/pnas/suppl/2017/05/03/1616736114.DCSupplemental/pnas.1616736114.sapp.pdf